opentracing-shim/src/main/java/io/opentelemetry/opentracingshim/SpanShim.java

```diff
@@ -214,10 +230,62 @@ static Attributes convertToAttributes(Map<String, ?> fields) {
       } else if (value instanceof Boolean) {
         attributesBuilder.put(booleanKey(key), (Boolean) value);
       } else {
-        attributesBuilder.put(stringKey(key), value.toString());
+        if (isExceptionEvent) {
+          switch (key) {
+            case Fields.ERROR_KIND:
+              key = "exception.type";
+              break;
+            case Fields.MESSAGE:
+              key = "exception.message";
+              break;
+            case Fields.STACK:
+              key = "exception.stacktrace";
+              break;
+            default:
+              break;
+          }
+        }
+        if (!isErrorEvent || !key.equals(Fields.ERROR_OBJECT) || !(value instanceof Throwable)) {
+          attributesBuilder.put(stringKey(key), value.toString());
+        }
       }
     }
 
     return attributesBuilder.build();
   }
+
+  private static Throwable findThrowable(Map<String, ?> fields) {
+    for (Map.Entry<String, ?> entry : fields.entrySet()) {
+      String key = entry.getKey();
+      Object value = entry.getValue();
+
+      if (value == null) {
```

Comment:
Don't think you need this since you have `value instanceof Throwable`

Reply:
removed